clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/RackAwareAssignor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.coordinator.group.api.assignor.ConsumerGroupPartitionAssignor;
+import org.apache.kafka.coordinator.group.api.assignor.GroupAssignment;
+import org.apache.kafka.coordinator.group.api.assignor.GroupSpec;
+import org.apache.kafka.coordinator.group.api.assignor.MemberAssignment;
+import org.apache.kafka.coordinator.group.api.assignor.PartitionAssignorException;
+import org.apache.kafka.coordinator.group.api.assignor.SubscribedTopicDescriber;
+import org.apache.kafka.coordinator.group.modern.MemberAssignmentImpl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The RackAwareAssignor is a consumer group partition assignor that takes into account the rack
+ * information of the members when assigning partitions to them.
+ * It needs all brokers and members to have rack information available.
+ */
+public class RackAwareAssignor implements ConsumerGroupPartitionAssignor {
```

Comment:
@dajac @FrankYang0529 @chia7712 is there a reason this assignor doesn't exist by default in the non-testing code? I commented here https://issues.apache.org/jira/browse/KAFKA-14450?focusedCommentId=18007549&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-18007549 that I believe KIP-848 introduced a feature regression by not supporting rack-aware assignment out of the box. It seems like quite the obvious miss, so I presume I may be missing something. Please let me know

Reply:
@FrankYang0529 Does it need a KIP? or the discussion is already included by another KIP?

Reply:
I think we can have PRs first. If it's too complicated, we can write a KIP to discuss it.

Reply:
> I think we can have PRs first. If it's too complicated, we can write a KIP to discuss it.

The out-of-box assignor is a kind of public API, right? I mean, users could configure it in the server properties

Reply:
We already have configuration `group.consumer.assignors` to support customized assignors. For KAFKA-19387, we will add rack-aware support to built-in assignors `RangeAssignor` and `UniformAssignor`.